lib/collection/query-param.js

```diff
@@ -59,6 +59,7 @@ _.assign(QueryParam.prototype, /** @lends QueryParam.prototype */ {
             key: _.get(param, 'key'), // we do not replace falsey with blank string since null has a meaning
             value: _.get(param, 'value')
         });
+        _.has(param, 'system') && (this.system = param.system);
```

Comment:
This will also include falsy `system` values, which is not what we want. Alternatively,
```js
param && param.system && (this.system = true);
```

Reply:
@kunagpal if we do that, there would be no way of unsetting a `system:true` value 🤔

Reply:
In that case, there should be a check to remove `system` if it is falsy, like in the original snippet.

Reply:
@kunagpal But that's what we are doing at other places as well
```js
// variable.js
options.hasOwnProperty('system') && (this.system = options.system);

// header.js
_.has(options, 'system') && (this.system = options.system);
```

All of these will go away once `Property` itself starts supporting this property.
We should keep consistency in the behaviour at all the places where this is used.